### PR TITLE
Hilbert Tutorial

### PR DESCRIPTION
**Pull request by SalahedeenIssa**
This isn't working on my end, though I believe it should work. Would love if someone could try out these notebooks and see if they work, or if they need to be modified. Fixes https://github.com/SupertechLabs/client-superstaq/issues/465

### CODE REVIEW

**Comment by SalahedeenIssa** [comment]
Btw -- when submitting via the client is it still necessary to set your options with the cq token ? @vtomole

**Comment by stephanielee9** [comment]
> Btw -- when submitting via the client (css specifically) is it still necessary to set your options with the cq token ? @vtomole

I think the answer is no, but I can check and I'll take a first pass at the notebooks

**Comment by stephanielee9** [comment]
@SalahedeenIssa Just want to make sure I'm running into the same issues as you:
* I can run the `qss` notebook fine
* I'm able to compile for the `css` notebook but not submit (I did not set `options`)

**Comment by SalahedeenIssa** [comment]
@stephanielee9 The qss notebook works for me too, and I can also compile using the css notebook, but regardless of whether I have options or not, I am unable to submit using css it seems. @vtomole Could you try submitting using css to see if this is just a me problem?

File: docs/source/optimizations/cq/cq_compile_css.ipynb [hunk 1]
```diff
@@ -5,7 +5,7 @@
    "id": "eb89a144",
```
Comment (stephanielee9) [inline docs/source/optimizations/cq/cq_compile_css.ipynb:5]: <p>Fine to keep as is but is there a reason the title is different (disregarding the word Cirq vs. Qiskit) between the <code>qss</code> and <code>css</code> notebooks?  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/527/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (stephanielee9) [inline docs/source/optimizations/cq/cq_compile_css.ipynb:5]: <p>These should be updated to the Markdown syntax for linked images. Not entirely sure if you need to fix what's already being implemented in https://github.com/Infleqtion/client-superstaq/pull/551/. Same for the <code>qss</code> notebook </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/527/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (stephanielee9) [inline docs/source/optimizations/cq/cq_compile_css.ipynb:5]: <p>This import should match the formatting of the other optimization notebooks. See: https://superstaq.readthedocs.io/en/latest/optimizations/qscout/qscout_css.html. Ditto for <code>qss</code> notebook </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/527/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (stephanielee9) [inline docs/source/optimizations/cq/cq_compile_css.ipynb:5]: <p>Nits:</p><ul><li>Update cirq-superstaq, Service(), and css.Service(api_key="token") to be in-line code</li><li>"ie" should be "i.e."</li><li>add link to https://docs-superstaq.readthedocs.io/en/stable/get_started/basics/basics_css.html#Set-up-access-to-Superstaq's-API for guide to setting environment variables.</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/527/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (stephanielee9) [inline docs/source/optimizations/cq/cq_compile_css.ipynb:5]: <p>Doesn't have to be very verbose but would be nice to have similar description and commentary for each of the codeblocks in this notebook so the user can have a better understanding of what's going on.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/527/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

File: docs/source/optimizations/cq/cq_compile_qss.ipynb [hunk 2]
```diff
@@ -21,26 +21,57 @@
     "</a>"
```
Comment (stephanielee9) [inline docs/source/optimizations/cq/cq_compile_qss.ipynb:21]: <p>Same comment as <code>css</code> notebook </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/527/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (stephanielee9) [inline docs/source/optimizations/cq/cq_compile_qss.ipynb:21]: <p>Update install instructions to match other <code>qss</code> optimization notebooks </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/527/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (stephanielee9) [inline docs/source/optimizations/cq/cq_compile_qss.ipynb:21]: <p>same comments as <code>css</code> notebook:</p><ul><li>make in-line code for qiskit-superstaq, qss.superstaq_provider.SuperstaQProvider(api_key="token")</li><li class="ql-indent-1">remove <code>api_key="token"</code></li><li>ie --&gt; i.e.</li><li>link to environment variable setup page</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/527/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (stephanielee9) [inline docs/source/optimizations/cq/cq_compile_qss.ipynb:21]: <p>same as <code>css</code> notebook - more description / commentary would be helpful here </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/527/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

**Comment by stephanielee9** [comment]
@SalahedeenIssa In addition to the notebook comments, it would be good to update:
* the `cq.rst` file to have a brief description (see https://superstaq.readthedocs.io/en/latest/optimizations/ibm/ibm.html) - doesn't need references but if there's a good cite, could be useful to the user
* It may be good to update all the instances of `cq` / ColdQuanta to Hilbert

Had no issue running any of the notebooks :)

File: docs/source/optimizations/cq/cq_compile_css.ipynb (hunk 1 above)
Comment (stephanielee9) [inline docs/source/optimizations/cq/cq_compile_css.ipynb:5]: <p>Can you restart and run all cells so the cells are numbered in order?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/527/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (SalahedeenIssa) [reply docs/source/optimizations/cq/cq_compile_css.ipynb:5]: <p>Ooh good catch - will fix the qiskit one</p>

Comment (SalahedeenIssa) [reply docs/source/optimizations/cq/cq_compile_css.ipynb:5]: <p>Done</p><p> </p>

Comment (SalahedeenIssa) [reply docs/source/optimizations/cq/cq_compile_css.ipynb:5]: <p>Done</p>

Comment (SalahedeenIssa) [reply docs/source/optimizations/cq/cq_compile_css.ipynb:5]: <p>Done</p>

File: docs/source/optimizations/hilbert/hilbert.rst [hunk 3]
```diff
@@ -0,0 +1,11 @@
+SuperstaQ for ColdQuanta
+========================
+For this tutorial, we will go over how to get started with using Superstaq to compile, optimize, and simulate circuits for the Infleqtion Hilbert backend. We will go over the process of setting up a Superstaq provider and compiling circuits to an Infleqtion  backend, retrieving the optimized circuits. By the end, we will provide a step-by-step guide to simulate a given circuit on Infleqtion's Hilbert simulator and get back the results!
```
Comment (stephanielee9) [inline docs/source/optimizations/hilbert/hilbert.rst:3]:
```suggestion
For this tutorial, we will go over how to get started with using Superstaq to compile, optimize, and simulate circuits for [Hilbert](https://www.infleqtion.com/coldquantalabs), Infleqtion's gate-based cloud-accessible neutral atom quantum computer. We will go over the process of setting up a Superstaq provider and compiling circuits to an Infleqtion backend, retrieving the optimized circuits. By the end, we will provide a step-by-step guide to simulate a given circuit on Infleqtion's Hilbert simulator and get back the results!
```

**Review by stephanielee9 — CHANGES_REQUESTED** [review]

File: docs/source/optimizations/cq/cq_compile_qss.ipynb (hunk 2 above)
Comment (stephanielee9) [reply docs/source/optimizations/cq/cq_compile_qss.ipynb]: The environment variable set up link is to the `css` page and should be the `qss` page

**Comment by stephanielee9** [comment]
@SalahedeenIssa A few things:
* Need to update `docs/source/index.rst` to switch the `cq` files to `hilbert` files
* Delete the `cq` directory from the `optimizations` directory
* Also my bad (I gave you the markdown syntax before): update the link in `hilbert.rst` to rst syntax: https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#links-to-external-web-pages
* I think the install instructions are a little out of sync with the other tutorials (e.g., for `qss` it should be `pip install -q qiskit-superstaq[examples]`)
* There seems to be description missing under the single and multiple circuit compilation
* The Binder and Colab links seem to still not be working for me

File: docs/source/optimizations/hilbert/hilbert_compile_css.ipynb [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+    "# Optional imports\n",
+    "import os  # used if setting a token as an environment variable"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ae0dfcb6",
+   "metadata": {},
+   "source": [
+    "To interface Superstaq via Cirq, we must first instantiate a service provider in ```cirq-superstaq``` with ```Service()```. We then supply a Superstaq API key (which you can get from https://superstaq.super.tech.) by either providing the API key as an argument of Service, i.e. : ```css.Service(api_key=\"token\")``` , or by setting it as an environment variable. (see more details [here](https://docs-superstaq.readthedocs.io/en/stable/get_started/basics/basics_css.html#Set-up-access-to-Superstaq's-API)) "
```
Comment (perlinm) [inline docs/source/optimizations/hilbert/hilbert_compile_css.ipynb:70]: super nit
```suggestion
    "To interface Superstaq via Cirq, we must first instantiate a service provider in ```cirq-superstaq``` with ```Service()```. We then supply a Superstaq API key (which you can get from https://superstaq.super.tech) by either providing the API key as an argument of Service, i.e., ```css.Service(api_key=\"token\")```, or by setting it as an environment variable. (see more details [here](https://docs-superstaq.readthedocs.io/en/stable/get_started/basics/basics_css.html#Set-up-access-to-Superstaq's-API)) "
```

File: docs/source/optimizations/hilbert/hilbert_compile_css.ipynb [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+   "source": [
+    "print(compiler_output.circuit)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "65c6a0f6",
+   "metadata": {},
+   "source": [
+    "If you would like to compile (or submit) on a different number of qubits, this can be done via the ```grid_shape``` option. This simply sets the shape of the rectangular qubit grid. However, specifying a grid that is incompatable with Hilbert's current capabilities will result in an error. "
```
Comment (perlinm) [inline docs/source/optimizations/hilbert/hilbert_compile_css.ipynb:168]: After some discussion with @vtomole: let's have the docs exclude `grid_shape` for now, but add it back once the [`grid_qubits` issue](https://github.com/Infleqtion/client-superstaq/issues/582#issuecomment-1614871528) is fixed

**Review by stephanielee9 — APPROVED** [review]
lgtm!

**Pull request merged**